Guide and minor main page changes for 1.1.0 (#609)
mongodb/terraform-provider-mongodbatlas: website/docs/guides/1.1.0-upgrade-guide.html.markdown

# MongoDB Atlas Provider 1.1.0: Upgrade and Information Guide

The Terraform MongoDB Atlas Provider version 1.1.0 has a number of new and exciting features and changes.

New Features:

> * Multi-cloud deployments can now be managed with [`mongodbatlas_advanced_clusters`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_clusters)
> * Invitations can now be managed with
> [`mongodbatlas_org_invitation`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/org_invitation) and [`mongodbatlas_project_invitation`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/project_invitation)
> * Projects created with [`mongodbatlas_project`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/project)` can now have a default owner set with `project_owner_id`
> * [`mongodbatlas_privatelink_endpoint`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/privatelink_endpoint) and [`mongodbatlas_privatelink_endpoint_service`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/privatelink_endpoint_service) now support Google Cloud'd Private Service Connect
> * Projects created with [`mongodbatlas_project`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/project)` can now have default alert creation disabled by setting `with_default_alerts_settings` to false.
> * [`mongodbatlas_search_index`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/search_index) now supports synonyms.
> * Atlas Serverless deployments, now in preview, can now be created with [`mongodbatlas_serverless_instance`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/serverless_instance)

Reviewer: passive -> active voice, periods at the end of list items that are complete sentences per [style guide](https://docs.mongodb.com/meta/style-guide/style/lists/#list-items), and minor updates for clarity
```suggestion
* You can now manage multi-cloud deployments with [`mongodbatlas_advanced_clusters`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_clusters).
* You can now manage invitations with
[`mongodbatlas_org_invitation`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/org_invitation) and [`mongodbatlas_project_invitation`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/project_invitation).
* You can now use `project_owner_id` to set a default owner of projects that you create with [`mongodbatlas_project`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/project).
* [`mongodbatlas_privatelink_endpoint`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/privatelink_endpoint) and [`mongodbatlas_privatelink_endpoint_service`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/privatelink_endpoint_service) now support private endpoints on Google Cloud with Private Service Connect.
* You can now set `with_default_alerts_settings` to `false` to disable the default alerts when you create projects with [`mongodbatlas_project`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/project).
* [`mongodbatlas_search_index`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/search_index) now supports synonyms.
* You can now create Atlas Serverless deployments (available as a preview feature) with [`mongodbatlas_serverless_instance`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/serverless_instance).
```

Changes:

> In addition, there are changes that will require modifications to your Terraform configuration between earlier versions of the provider and this one.

Reviewer: The bulk of these changes are covered in the next section. I think it's appropriate to link there.

consider something like:

Version 1.10 removes|deprecates the following resources:

- [`mongodbatlas_cloud_provider_snapshot`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_snapshot)
- [`mongodbatlas_cloud_provider_snapshot_restore_job`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_snapshot_restore_job)

If your Terraform configuration uses these resources, see [Migration from Cloud Provider Snapshot to Cloud Backup Snapshot](#migration-from-cloud-provider-snapshot-to-cloud-backup-snapshot) to migrate to the new resources **before you upgrade**.

Author: Great suggestion!  I just added a clarification that they will be removed in a future major version.

* `mongodbatlas_cloud_backup_snapshot`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_backup_snapshot) replaces the now deprecated `mongodbatlas_cloud_provider_snapshot`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_snapshot).
* `mongodbatlas_cloud_backup_snapshot_restore_job`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_backup_snapshot_restore_job) replaces the now deprecated `mongodbatlas_cloud_provider_snapshot_restore_job`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_snapshot_restore_job).

1.1.0 also includes general improvements and bug fixes. See the [CHANGELOG](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CHANGELOG.md) for more specific information.

## Migration from Cloud Provider Snapshot to Cloud Backup Snapshot

> The term Cloud Provider has been deprecated in favor of the term Cloud Backup.  In order to support this change within the Terraform provider, related resources and data sources with the older terminology have been deprecated and new resources and data sources added to support the new terminology.  In order to upgrade from the previous resources to the new you'll need to select from one of the following two options:

Reviewer: bold to emphasize terms
```suggestion
The term **Cloud Provider** has been deprecated in favor of the term **Cloud Backup**.  In order to support this change within the Terraform provider, related resources and data sources with the older terminology have been deprecated and new resources and data sources have been added to support the new terminology.  In order to upgrade from the previous resources to the new you'll need to select from one of the following two options:
```

Reviewer: have these been deprecated or removed? Based on the steps below, looks like the old providers have been removed.

Author: Just deprecated but obviously that's not clear.  Will adjust.

> 

Reviewer: the difference between option 1 and option 2 is not clear to me. it looks like maybe if users have used the old providers, they must complete option 2. if they don't they can use option 1. is this accurate?

Author: One requires just changing the names in the state file. Not a great practice in the Terraform world but works. The second is the more official way, but slightly harder.   It requires fully removing the old resources from state and then re-importing them.

### Option 1: Replace the name of resource in terraform state

> 1. Replace the name of the deprecated provider `mongodbatlas_cloud_provider_snapshot` with the new name `mongodbatlas_cloud_backup_snapshot` in the Terraform state file (`terraform.tfstate`).  Do the same for `mongodbatlas_cloud_provider_snapshot_restore_job` and `mongodbatlas_cloud_backup_snapshot_restore_job`

Reviewer: ```suggestion
1. Replace the name of the deprecated resource `mongodbatlas_cloud_provider_snapshot` with the new name `mongodbatlas_cloud_backup_snapshot` in the Terraform state file (`terraform.tfstate`).  Do the same for `mongodbatlas_cloud_provider_snapshot_restore_job` and `mongodbatlas_cloud_backup_snapshot_restore_job`.
```

>    **NOTE** Before performing any Terraform state change please backup your state.

Reviewer: ```suggestion
   **NOTE** Before performing any Terraform state change, please backup your state.
```

> 2. Upgrade the provider to v1.1.0

Reviewer: ```suggestion
2. Upgrade the provider to v1.1.0.
```

> 3. Run a `terraform plan` to ensure no changes are detected

Reviewer: ```suggestion
3. Run a `terraform plan` to ensure no changes are detected.
```

```bash
terraform plan

(...)

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences,
so no changes are needed.
```

> ### Option 1:  Remove the existing and import the new resource

Reviewer: choose deprecated or removed based on if the old resources are removed
```suggestion
### Option 2:  Remove the deprecated|removed and import the new resources
```

Author: Deprecated is accurate, so choosing that.

> 1. Remove your existing `mongodbatlas_cloud_provider_snapshot` and `mongodbatlas_cloud_provider_snapshot_restore_job` resources from the Terraform state file
>    **NOTE** Before performing any Terraform state change please backup your state.

Reviewer: ```suggestion
1. Remove your existing `mongodbatlas_cloud_provider_snapshot` and `mongodbatlas_cloud_provider_snapshot_restore_job` resources from the Terraform state file.
   **NOTE** Before performing any Terraform state change, please backup your state.
```

```bash
$ terraform state rm mongodbatlas_cloud_provider_snapshot.test
$ terraform state rm mongodbatlas_cloud_provider_snapshot_restore_job.test

```

> 2. Upgrade the provider to v1.1.0

Reviewer: ```suggestion
2. Upgrade the provider to v1.1.0.
```

3. Change your Terraform configuration file, replacing `mongodbatlas_cloud_provider_snapshot` with ``mongodbatlas_cloud_backup_snapshot` and `mongodbatlas_cloud_provider_snapshot_restore_job` with `mongodbatlas_cloud_backup_snapshot_restore_job`

> 4. Import your existing `mongodbatlas_cloud_provider_snapshot` to `mongodbatlas_cloud_backup_snapshot` and `mongodbatlas_cloud_provider_snapshot_restore_job` to `mongodbatlas_cloud_backup_snapshot_restore_job`

Reviewer: ```suggestion
4. Import your existing `mongodbatlas_cloud_provider_snapshot` to `mongodbatlas_cloud_backup_snapshot` and `mongodbatlas_cloud_provider_snapshot_restore_job` to `mongodbatlas_cloud_backup_snapshot_restore_job`:
```

```bash
    terraform import mongodbatlas_cloud_backup_snapshot.test <project_id>-<cluster_name>-<snapshot_id>
```
```bash
    terraform import mongodbatlas_cloud_backup_snapshot_restore_job.test <project_id>-<cluster_name>-<job_id>
```

> 5. Run a `terraform plan` to ensure no changes are detected

Reviewer: ```suggestion
5. Run a `terraform plan` to ensure no changes are detected.
```

# MongoDB Atlas Provider 1.1.0: Upgrade and Information Guide

The Terraform MongoDB Atlas Provider version 1.1.0 has a number of new and exciting features and changes.

New Features:

> * You can now manage multi-cloud deployments with [`mongodbatlas_advanced_clusters`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_clusters).

Reviewer: These links will work when this version is released, correct?

Author: Correct, they won't work till the release goes out.

   **NOTE** Before performing any Terraform state change, please backup your state.

2. Upgrade the provider to v1.1.0.

3. Run a `terraform plan` to ensure no changes are detected.

```bash
terraform plan

(...)

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences,
so no changes are needed.
```

### Option 2:  Remove the deprecated and import the new resources

1. Remove your existing `mongodbatlas_cloud_provider_snapshot` and `mongodbatlas_cloud_provider_snapshot_restore_job` resources from the Terraform state file.
   **NOTE** Before performing any Terraform state change, please backup your state.

```bash
$ terraform state rm mongodbatlas_cloud_provider_snapshot.test
$ terraform state rm mongodbatlas_cloud_provider_snapshot_restore_job.test

```

2. Upgrade the provider to v1.1.0.

> 3. Change your Terraform configuration file, replacing `mongodbatlas_cloud_provider_snapshot` with ``mongodbatlas_cloud_backup_snapshot` and `mongodbatlas_cloud_provider_snapshot_restore_job` with `mongodbatlas_cloud_backup_snapshot_restore_job`

Reviewer: minor formatting and punctuation changes
```suggestion
3. Change your Terraform configuration file, replacing `mongodbatlas_cloud_provider_snapshot` with `mongodbatlas_cloud_backup_snapshot` and `mongodbatlas_cloud_provider_snapshot_restore_job` with `mongodbatlas_cloud_backup_snapshot_restore_job`.
```
